compiler/rustc_middle/src/ty/context.rs

```diff
@@ -525,6 +525,9 @@ impl<'tcx> TyCtxt<'tcx> {
     pub fn feed_local_def_id(self, key: LocalDefId) -> TyCtxtFeed<'tcx, LocalDefId> {
         TyCtxtFeed { tcx: self, key }
     }
+    pub fn feed_owner_id(self, key: hir::OwnerId) -> TyCtxtFeed<'tcx, hir::OwnerId> {
```

Comment:
I'd prefer for this to be a map function from a `TyCtxtFeed<LocalDefId>`. The above function is a too big footgun already and should be documented with lots of warnings and a scary name.

Reply:
I addressed the documentation part in https://github.com/rust-lang/rust/pull/121080 (also adds a sanity assertion)

Reply:
Yeah, I've just wanted to ask why it's a footgun, I've been using it pretty liberally recently.

Reply:
Basically it creates a situation where query A creates DefId B, then query C feeds query D for B. In the next incremental invocation, A is rerun, but C is loaded from the cache (thus D's value is never re-fed for B). If `D` is not cached, we now invoke the body of `D` for `B`, which usually just ICEs.

There are worse situations where we end up with outdated values.

Only queries that create a def id can reliably feed queries for that id.

Ideally we'd use `create_def` in the resolver and keep the entire `TyCtxtFeed<LocalDefId>` around in the tables instead of just the `LocalDefId`, that would avoid having to manually create the `TyCtxtFeed` again, but that's not a trivial refactoring.